pkg/kubelet/kubelet.go

```diff
@@ -2968,6 +2968,21 @@ func (kl *Kubelet) handlePodResourcesResize(pod *v1.Pod, podStatus *kubecontaine
 			kl.statusManager.SetPodResizeInProgressCondition(pod.UID, "", "", false)
 		} else {
 			// (Allocated == Actual) => clear the resize in-progress status.
+			// If pod resize completed, print ResizeCompleted event
+			resizeStatus := kl.statusManager.GetPodResizeConditions(allocatedPod.UID)
```

Comment:
The pod resize conditions in the status manager are not backfilled on kubelet restart today, so if the kubelet restarts while a pod resize is in progress, the status manager conditions will be empty. Then if the resize completes, the event will not be emitted. 

https://github.com/kubernetes/kubernetes/pull/131157 adds backfilling the conditions to the status manager, which would make this no longer an issue, but that PR is not (yet) merged. Maybe keep an eye on it